### PR TITLE
removed deprecated code from s3save

### PR DESCRIPTION
**Pull request by JackStat**
I was receiving the deprecating warnings with this function so I replace the deprecated functions with the more current versions.

### CODE REVIEW

**Comment by leeper** [comment]
Thanks. I'll merge this in a little bit.

**Comment by JackStat** [comment]
I should have waited on the pull request.  Give me some time

**Comment by leeper** [comment]
Sounds good. Can you do two things for me before you finish everything:
- [ ] bump the version number in DESCRIPTION
- [ ] add yourself to the `Authors@R` field in DESCRIPTION

**Comment by JackStat** [comment]
Alright, Some of the changes were a bit large but I got the sense that the package was built to work well for us-east-1 and so I made some modifications to make it easier to use other zones.  I hope it makes sense but if it doesn't let me know what I need to change.

File: R/http.r [hunk 1]
```diff
@@ -92,7 +115,7 @@ s3HTTP <- function(verb = "GET",
     }
     
     if (verb == "GET") {
-      r <- httr::GET(url, H, query = query, ...)
```
Comment (leeper) [inline R/http.r]: Why delete the passing of `...` here?

File: R/http.r [hunk 2]
```diff
@@ -35,27 +35,50 @@ s3HTTP <- function(verb = "GET",
...
+      if(Sys.getenv("AWS_DEFAULT_REGION") == ""){
+        
+      }
+      if (bucketname != "") {
+        url <- paste0("https://", bucketname, ".s3-", region, ".amazonaws.com")
       } else {
         url <- paste0("https://s3-", region, ".amazonaws.com")
       }
     }
+    message(url)
```
Comment (leeper) [inline R/http.r]: Please delete this `message()` call.

**Comment by JackStat** [comment]
Nice catches.  Sorry about that.  I thought that I put those back

**Comment by leeper** [comment]
This generally looks good. I'm not sure about some of the internal handling of `region` in `s3HTTP()` so I'll need to take another look later. But in the meantime, thank you very much for the contribution!

**Comment by JackStat** [comment]
Happy to help and learn. If there is a better way that comes to mind let's do it.

**Comment by leeper** [comment]
@JackStat Thanks again for this! As you'll see, I've [squashed your commits](https://github.com/cloudyr/aws.s3/commit/77316977d8946309bbc07780bf807042faa57511) and incorporated them, though I [made some further changes](https://github.com/cloudyr/aws.s3/commit/9ed2866dad7e7f06024f4533155751ae6bed304e) in terms of handling regions inside `s3HTTP()`.

You'll want to rebase your branch to be on track with the updated repo.

**Comment by JackStat** [comment]
I like how your created the get_region function. Nice work and thanks for the effort on this package.

**Pull request closed without merging**